### PR TITLE
feat(activity-center): Add details view dialog for contact requests

### PR DESCRIPTION
**Pull request by borismelnik**
Fixes: #8461

### What does the PR do

Details view popup for Contact Requests

### Affected areas

Activity Center

### Screenshot of functionality (including design for comparison)

- [ ] I've checked the design and this PR matches it

https://user-images.githubusercontent.com/82511785/205679961-8ec2c404-be76-42db-88cc-cf466e1b3b05.mov

### CODE REVIEW

File: ui/app/mainui/activitycenter/popups/ReviewContactRequestPopup.qml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    signal accepted
+    signal declined
+
+    anchors.centerIn: parent
+    width: 638
+    padding: Style.current.bigPadding
+
+
+    property StatusMessageDetails messageDetails: StatusMessageDetails {
```
Comment (jrainville) [inline ui/app/mainui/activitycenter/popups/ReviewContactRequestPopup.qml:28]: Is it necessary to construct this component here? Do we reuse those properties a lot or we could just set the values directly in the right places below?

**Review by jrainville — COMMENTED** [review]
Good job. I just have small questions and suggestions

**Comment by caybro** [comment]
Can you pls fix what's already obvious from the video you posted? :) The text overflowing into the accept/reject buttons

**Comment by jrainville** [comment]
> Can you pls fix what's already obvious from the video you posted? :) The text overflowing into the accept/reject buttons

@MishkaRogachev might already be fixing that. The reason I assigned this to Boris even though Mishka is the one on all the AC tasks is because of the low chance of conflict.

File: ui/app/mainui/activitycenter/popups/ReviewContactRequestPopup.qml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import StatusQ.Popups 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+import StatusQ.Core.Utils 0.1 as CoreUtils
+
+import shared 1.0
+import utils 1.0
+
+StatusModal {
```
Comment (caybro) [inline ui/app/mainui/activitycenter/popups/ReviewContactRequestPopup.qml:15]: StatusDialog? StatusModal is deprecated

File: ui/app/mainui/activitycenter/popups/ReviewContactRequestPopup.qml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                    Layout.alignment: Qt.AlignVCenter
+                    Layout.fillWidth: true
+                    Layout.rightMargin: -Style.current.halfPadding
+                }
+            }
+        }
+    }
+    rightButtons: [
+        StatusButton {
+            text: "Accept Contact Request"
```
Comment (caybro) [inline ui/app/mainui/activitycenter/popups/ReviewContactRequestPopup.qml:98]: qsTr()

File: ui/app/mainui/activitycenter/popups/ReviewContactRequestPopup.qml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            onClicked: {
+                root.accepted()
+                root.close()
+            }
+        },
+        StatusButton {
+            type: StatusBaseButton.Type.Danger
+            text: "Reject Contact Request"
```
Comment (caybro) [inline ui/app/mainui/activitycenter/popups/ReviewContactRequestPopup.qml:106]: qsTr()

File: ui/app/mainui/activitycenter/views/ActivityNotificationContactRequest.qml [hunk 5]
```diff
@@ -33,5 +34,13 @@ ActivityNotificationMessage {
             root.store.contactsStore.dismissContactRequest(senderId)
             root.store.contactsStore.blockContact(senderId)
         }
+        onDetailsClicked: reviewRequestPopup.open()
+    }
+
+    ReviewContactRequestPopup {
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/mainui/activitycenter/views/ActivityNotificationContactRequest.qml:40]: I'd say yes; we might want to (re)use it from the Profile Dialog at some point

**Review by caybro — CHANGES_REQUESTED** [review]

**Review by MishkaRogachev — CHANGES_REQUESTED** [review]
Screenshot 2022-12-06 at 11 24 58

The text here is cropped with CTA component

**Comment by borismelnik** [comment]
Thank you guys, will fix it asap

**Comment by borismelnik** [comment]
Снимок экрана 2022-12-06 в 21 30 39

I also fixed it cos it one line fix.

**Review by MishkaRogachev — APPROVED** [review]

**Review by jrainville — APPROVED** [review]

**Pull request merged**